ids/id.go

```diff
@@ -32,6 +32,12 @@ var (
 // ID wraps a 32 byte hash used as an identifier
 type ID [IDLen]byte
 
+// Any modification to Bytes will be lost since id is passed-by-value
+// Use Writable(ID) is you need to modify the ID
+func (id ID) Bytes() []byte {
```

Comment:
we do Bytes method with NodeID, I think we should have it for ID as well, or drop both of them